Makefile

```diff
@@ -85,4 +85,4 @@ docker-push:
 # Marking this as PHONY so it will build every time (so you always have the correct version locally)
 .PHONY: $(CONTROLLER_GEN)
 $(CONTROLLER_GEN):
-	go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
+	cd hack/tools && go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
```

Comment:
bin ends up in hack/tools/, not top level directory. I ended up make not able to find `bin/controller-gen`

Reply:
I would recommend this change:

```
CONTROLLER_GEN := hack/tools/bin/controller-gen
```

Reply:
wait no this is completely wrong

Reply:
Updated recommendation:

```
TOOLS_DIR := hack/tools
CONTROLLER_GEN := bin/controller-gen
CONTROLLER_GEN_PATH := $(TOOLS_DIR)/$(CONTROLLER_GEN)
```

```
# Build controller-gen
$(CONTROLLER_GEN_PATH): $(TOOLS_DIR)/go.mod
	cd $(TOOLS_DIR) && go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
```

I'm kind of making stuff up here with removing the .PHONY but idk, might be cool?

Also changing all the CONTROLLER_GEN references to CONROLLER_GEN_PATH

Reply:
Updated with a slight variation on your suggestion - thanks @chuckha. PTAL!